maltoolbox/attackgraph/node.py

```diff
@@ -89,28 +97,20 @@ def __deepcopy__(self, memo) -> AttackGraphNode:
             return memo[id(self)]
 
         copied_node = AttackGraphNode(
-            type=self.type,
-            lang_graph_attack_step=self.lang_graph_attack_step,
-            name=self.name,
-            ttc=self.ttc,
-            id=self.id,
-            asset=self.asset,
-            children=[],
-            parents=[],
-            detectors=self.detectors,
-            defense_status=self.defense_status,
-            existence_status=self.existence_status,
-            is_viable=self.is_viable,
-            is_necessary=self.is_necessary,
-            compromised_by=[],
-            tags=set(),
-            attributes={},
-            extras={},
+            node_id = self.id,
+            model_asset = self.model_asset,
+            lg_attack_step = self.lg_attack_step
         )
 
         copied_node.tags = copy.deepcopy(self.tags, memo)
-        copied_node.attributes = copy.deepcopy(self.attributes, memo)
         copied_node.extras = copy.deepcopy(self.extras, memo)
+        copied_node.ttc = copy.deepcopy(self.ttc, memo)
+        copied_node.detectors = copy.deepcopy(self.detectors, memo)
```

Comment:
These  (except `extras`) should not change after the node is created and they will already be set during node creation (in __init__) right? Is there an explicit need to copy them over here?

Reply:
Good catch, some of the code changed and I forgot to update it. I will still copy over the `tags` and `ttc`. I think we should assume that they could be modified from the language default, but the others can be skipped.